### DISCUSSION TITLE
How to start docker with flags -it using CliWrap?

### DISCUSSION THREAD

**Discussion by miroljub1995** [Tyrrrz/CliWrap#160] [Q&A]
When I want to start `docker run -it alpine cat` it gives me:
```the input device is not a TTY```. I'm running the following code using `dotnet run` in wsl:
```cs
await using Stream stdOut = Console.OpenStandardOutput();
await using Stream stdErr = Console.OpenStandardError();
await (CliWrap.Cli.Wrap("docker") | (stdOut, stdErr))
    .WithArguments($"run -it alpine cat")
    .ExecuteAsync();
```
Any idea how to start it?
Thanks.

**Comment by Tyrrrz** [comment]
Wrap it in an actual TTY:

```csharp
await using Stream stdOut = Console.OpenStandardOutput();
await using Stream stdErr = Console.OpenStandardError();
await (Cli.Wrap("cmd") | (stdOut, stdErr))
    .WithArguments(new[] {"/c", "docker run -it alpine cat"})
    .ExecuteAsync();
```

**Reply by miroljub1995** [reply]
Now it says `the input device is not a TTY.  If you are using mintty, try prefixing the command with 'winpty'`

Not sure how, but default dotnet Process works,
```cs
System.Diagnostics.Process.Start("docker", "run -it redis redis-cli")
```

Do you know what is the difference in running with CliWrap and Process in this case, is it only setup of ProcessStartInfo?

**Reply by Tyrrrz** [reply]
> Do you know what is the difference in running with CliWrap and Process in this case, is it only setup of ProcessStartInfo?

The stdout/stderr/stdin are redirected, so they don't reach an actual terminal. I thought running via `cmd` would be enough to fool it because the streams are not redirected at the outer layer.

Anyway, maybe you can find some useful answers here: https://stackoverflow.com/questions/43099116/error-the-input-device-is-not-a-tty
